### PR TITLE
Tukaiz develop

### PR DESCRIPTION
**Pull request by francirp**

### CODE REVIEW

**Comment by francirp** [comment]
Merging so I can push to production. Will monitor for comments

File: app/assets/javascripts/locomotive/views/pages/edit_view.js.coffee [hunk 1]
```diff
@@ -2,39 +2,3 @@ Locomotive.Views.Pages ||= {}
 
 class Locomotive.Views.Pages.EditView extends Locomotive.Views.Pages.FormView
 
-  save: (event) ->
```
Comment (bhennes2) [inline app/assets/javascripts/locomotive/views/pages/edit_view.js.coffee:5]: Is the only way to save now through the preview option? Is there a way to just disable the binding of the save action to the form submitting vs. deleting the save function? I'd hate to remove this save function as we might want to use it in other places.

File: app/controllers/locomotive/api/memberships_controller.rb [hunk 2]
```diff
@@ -13,7 +13,7 @@ def show
       end
 
       def create
-        @membership.from_presenter(params[:membership].merge(role: Locomotive::Ability::DEFAULT_ROLE)) # force author by default
+        @membership.from_presenter(params[:membership])
```
Comment (bhennes2) [inline app/controllers/locomotive/api/memberships_controller.rb:16]: How are we setting the default role now? In the model?

File: app/models/locomotive/preview.rb [hunk 3]
```diff
@@ -0,0 +1,14 @@
+module Locomotive
+  class Preview
+    include Locomotive::Mongoid::Document
+
+    ## fields ##
+    field :page_id
+    field :params
```
Comment (bhennes2) [inline app/models/locomotive/preview.rb:7]: Do we also need a :site_id field?

File: app/views/locomotive/pages/edit.html.haml [hunk 4]
```diff
@@ -11,8 +11,9 @@
 
 %p!= t('.help')
 
-= semantic_form_for @page, url: page_path(@page), html: { multipart: true } do |form|
+/ = semantic_form_for @page, url: page_previews_path(@page), method: :post, html: { multipart: true } do |form|
```
Comment (bhennes2) [inline app/views/locomotive/pages/edit.html.haml:14]: Let's just remove this line if you commented it out.

File: lib/locomotive/liquid/tags/nav.rb [hunk 5]
```diff
@@ -59,6 +71,7 @@ def render(context)
             list_class  = !@options[:class].blank? ? %( class="#{@options[:class]}") : ''
             output      = %{<nav id="#{@options[:id]}"#{list_class}><ul>\n#{output}</ul></nav>}
           end
+          output.prepend(%{<li class="parent #{'active' if top_page_active?}"><a href="/#{top_page.fullpath}">#{top_page.title}</a></li>})
```
Comment (bhennes2) [inline lib/locomotive/liquid/tags/nav.rb:74]: Should we make the string "active" a constant or configurable options in the @options hash?

Comment (bhennes2) [inline lib/locomotive/liquid/tags/nav.rb:74]: Maybe @options[:active_class]?

File: lib/locomotive/render.rb [hunk 6]
```diff
@@ -18,14 +18,12 @@ def render_locomotive_page(path = nil, assigns = {})
       if request.fullpath =~ /^\/#{Locomotive.mounted_on}\//
         render template: '/locomotive/errors/404', layout: '/locomotive/layouts/not_logged_in', status: :not_found
       else
-        @page ||= self.locomotive_page(path)
-
-        if @page.present? && @page.redirect?
+        @page ||= (assigns[:page] || self.locomotive_page(path))
+        should_redirect = assigns[:no_redirect].present? ? !assigns[:no_redirect] : true
```
Comment (bhennes2) [inline lib/locomotive/render.rb:22]: I think you can just say this:

```
should_redirect = assigns[:no_direct].nil?
```

File: app/assets/javascripts/locomotive/views/pages/edit_view.js.coffee (hunk 1 above)
Comment (francirp) [inline app/assets/javascripts/locomotive/views/pages/edit_view.js.coffee:5]: This code only applies to the pages#edit action...which we won't be needing the JS save for anymore

File: app/controllers/locomotive/api/memberships_controller.rb (hunk 2 above)
Comment (francirp) [inline app/controllers/locomotive/api/memberships_controller.rb:16]:
``` ruby
# membership.rb
field :role, default: Locomotive::Ability::DEFAULT_ROLE
```

File: app/models/locomotive/preview.rb (hunk 3 above)
Comment (francirp) [inline app/models/locomotive/preview.rb:7]: I actually don't know the answer to that. It seems as if the belongs_to with MongoDB takes care of the foreign key automatically. The Page class for example belongs_to site and does not have a field for site_id. It does, however, have the following: 

``` ruby
index site_id: 1
```

As I build out the previews more over the next few days, I should have an answer for you.

File: lib/locomotive/render.rb (hunk 6 above)
Comment (francirp) [inline lib/locomotive/render.rb:22]: Good call

File: lib/locomotive/liquid/tags/nav.rb (hunk 5 above)
Comment (francirp) [inline lib/locomotive/liquid/tags/nav.rb:74]: I think this is a good idea and may be part of a larger refactoring to the Nav class if change presents itself (as you will see in the new pull request, I have some more classes hard coded in...not ideal)

**Pull request merged**